packages/devtools/lib/src/charts/flame_chart_canvas.dart

```diff
@@ -428,17 +367,19 @@ class FlameChartNode {
 
   final Color selectedTextColor;
 
-  final CpuStackFrame stackFrame;
+  final T data;
+
+  DisplayTextProvider displayTextProvider;
 
   final bool rounded;
 
   final Map<String, num> textMeasurements = {};
 
   Rect rect;
 
-  String get text => stackFrame.name;
+  String get text => displayTextProvider(data);
 
-  String get tooltip => stackFrame.toString();
+  String get tooltip => data.toString();
```

Comment:
[optional] Shorter, and may highlight nicer, as `String get tooltip => '$data';`